### DISCUSSION TITLE
Help on using macvlan mode in Docker Desktop when there is no network inside the container

### DISCUSSION THREAD

**Discussion by 1115714829** [microsoft/WSL#9825] [Q&A]
Is the network in WSL2 Full One (NAT1) without using the docker macvlan mode?

I installed Docker Desktop on my Windows 11.

My WSL2 is configured to bridge to the specified network card under the. wslconfig folder of the Windows user folder. The internal environment of WSL2 can obtain the IP address assigned by my superior routing DHCP.

[wsl2]

networkingMode=bridged

vmSwitch=WSLlan

Docker internally installs ubuntu (because I need to use the original Linux kernel) before deploying the application.

However, if I use the Macvlan mode to link to WSL2, there will be a problem with no network inside the container.

The reason why I don't use Hyper is that WSL can directly share 16G of memory with Windows and use it on my computer. Usually, my system uses 8G+of memory. If I directly configure 8G of memory for Hyper, sometimes it will cause other applications I open to fail to start due to insufficient memory.

**Comment by leosamuele221** [comment]
Same problem here.
i added 
`[wsl2]
networkingMode=bridged
vmSwitch=WSLlan`
to my .wslconfig and now wsl2 is a physical device inside the network.
But when i run a container in network_mode: host (with docker-compose) OR i use a macvlan network there is no connection with the container
